src/checkers/inference/InferenceOptions.java

```diff
@@ -0,0 +1,356 @@
+        }
+
+        if (typesystem != null) {
+            TypeSystemSpec spec = typesystems.get(typesystem.toLowerCase());
+            if (spec == null) {
+                errors.add("Unrecognized typesystem.  Current typesystems:\n"
+                           + PluginUtil.join("\n", typesystems.keySet()));
+            } else {
+                spec.apply();
+            }
+        }
+
+        if (checker == null) {
+            errors.add("You must specify exactly one checker using --checker!");
+        }
+
+        if (mode == null) {
+            if (requireMode) {
+                errors.add("You must specify a mode of operation using -m or --mode");
+            }
+
+        } else {
+            mode = mode.toUpperCase();
+
+            Mode modeEnum = null;
+            try {
+                modeEnum = Mode.valueOf(InferenceOptions.mode);
+
+            } catch (IllegalArgumentException iexc) {
+                System.out.println("Could not recognize mode: " + InferenceOptions.mode + "\n"
+                        + "valid modes: " + PluginUtil.join(", ", Mode.values()));
+                System.exit(1);
+            }
+
+            if (modeEnum != Mode.TYPECHECK) {
+                if (solver == null) {
+                    if (jsonFile != null) {
+                        solver = JsonSerializerSolver.class.getCanonicalName();
+                        if (solverArgs == null || solverArgs.isEmpty()) {
+                            solverArgs = "constraint-file=" + InferenceOptions.jsonFile;
+                        } else {
+                            solverArgs = solverArgs + "," + "constraint-file=" + InferenceOptions.jsonFile;
+                        }
+                    } else {
+                        errors.add("You must specify a solver using --solver or a --jsonFile to write constraints in.");
+                    }
+                } else if (jsonFile != null) {
+                    errors.add("You may specify EITHER a solver or jsonFile but not both!");
+                }
+            }
+
+            if (modeEnum.ordinal() >= Mode.ROUNDTRIP.ordinal()) {
+                if (afuOutputDir == null) {
+                    if (!inPlace) {
+                        errors.add("You must specify an Annotation File Utilities output directory (--afuOutputDir or -afud) or --inPlace.");
+                    }
+                } else if (inPlace) {
+                    errors.add("You cannot specify both an Annotation File Utilities output directory (--afuOutputDir or -afud) and --inPlace.");
+                }
+
+                if (afuOptions != null && afuOptions.contains("\\s-d\\s")) {
+                    errors.add("Annotation File Utilities output dir must be specified via (--afuOutputDir or -afud) not -d in AFU Options.");
+                }
+            }
+        }
+
+        return new InitStatus(options, errors, help);
+    }
+
+    public static final Map<String, TypeSystemSpec> typesystems = new LinkedHashMap<>();
```

Comment:
I have locally, that's why I hadn't sent the review out yet as I wanted to make a first pass.  But since we don't have much time before the first meeting, I figured I might as well just send out the link.